File: 30DaysOfCode/day23_bstLevelOrderTraversal/bstLevelOrderTraversal.php
<?php
class Node {
    public $left, $right;
    public $data;
    function __construct($data) {
        $this->left = $this->right = null;
        $this->data = $data;
    }
}

class Solution {
    public function insert($root, $data) {
        if ($root == null) {
            return new Node($data);
        } else {            
            if ($data <= $root->data) {
                $cur = $this->insert($root->left, $data);
                $root->left = $cur;
            } else {
                $cur = $this->insert($root->right, $data);
                $root->right = $cur;
            }
            return $root;
        }
    }
    
    public function levelOrder($root) {
        $q = array();
        $q[] = $root;
        
        while (sizeof($q) > 0) {
            $firstElement = array_shift($q);
            echo $firstElement->data . " ";
            
            if ($firstElement->left != null)
                $q[] = $firstElement->left;
            if ($firstElement->right != null)
                $q[] = $firstElement->right;
        }
    }
}

$myTree = new Solution();
$root = null;
$T = intval(fgets(STDIN));
while ($T-- > 0) {
    $data = intval(fgets(STDIN));
    $root = $myTree->insert($root, $data);
}
$myTree->levelOrder($root);
?>

File: 30DaysOfCode/day23_bstLevelOrderTraversal/bstLevelOrderTraversal.rb
class Node
    attr_accessor :left, :right, :data
    def initialize data
        @left = @right = nil
        @data = data
    end
end

class Solution
    def insert (root, data)
        if root == nil
            return Node.new(data)
        else
            if data <= root.data
                cur = self.insert(root.left, data)
                root.left = cur
            else
                cur = self.insert(root.right, data)
                root.right = cur
            end
        end
        
        return root
    end
    
    def levelOrder(root)
        @q = []
        @q.push(root)
        
        while (@q.length > 0) do
            first_node = @q.shift
            print(first_node.data, " ")
            
            if (first_node.left != nil)
                @q.push(first_node.left)
            end
            if (first_node.right != nil)
                @q.push(first_node.right)
            end
        end
    end
end

myTree = Solution.new
root = nil
T = gets.to_i

for i in 1..T
    data = gets.to_i
    root = myTree.insert(root, data)
end

myTree.levelOrder(root)
